src/main/kotlin/org/rust/ide/refactoring/introduceConstant/ui.kt

```diff
@@ -0,0 +1,157 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.refactoring.introduceConstant
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.editor.colors.EditorColors
+import com.intellij.openapi.editor.colors.EditorColorsManager
+import com.intellij.openapi.editor.markup.HighlighterLayer
+import com.intellij.openapi.editor.markup.HighlighterTargetArea
+import com.intellij.openapi.editor.markup.MarkupModel
+import com.intellij.openapi.editor.markup.RangeHighlighter
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.popup.JBPopupFactory
+import com.intellij.openapi.ui.popup.JBPopupListener
+import com.intellij.openapi.ui.popup.LightweightWindowEvent
+import com.intellij.openapiext.isUnitTestMode
+import com.intellij.psi.PsiElement
+import org.jetbrains.annotations.TestOnly
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.block
+import java.awt.Component
+import javax.swing.DefaultListCellRenderer
+import javax.swing.JList
+
+class Highlighter(private val expr: RsExpr, private val project: Project, private val editor: Editor) : JBPopupListener {
+    private val factory = RsPsiFactory(project)
+    private var highlighter: RangeHighlighter? = null
+    private var comment: PsiElement? = null
+    private val attributes = EditorColorsManager.getInstance().globalScheme.getAttributes(EditorColors.SEARCH_RESULT_ATTRIBUTES)
+
+    fun onSelect(candidate: InsertionCandidate) {
+        dropHighlighter()
+        val markupModel: MarkupModel = editor.markupModel
+
+//        val newComment = factory.createComment("<here>")
```

Comment:
Well, let's use the usual IntelliJ UI for this for now and drop these comments.

Highlight a position with comments is an interesting idea. Something like inlay hints maybe even better :thinking: